src/main/java/de/melanx/aiotbotania/items/terrasteel/ItemTerraSteelAIOT.java

```diff
@@ -0,0 +1,480 @@
+                        }
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        public List<BlockPos> adjacent(BlockPos original) {
+            List<BlockPos> coords = new ArrayList<>();
+
+            for (int dx = -1; dx <= 1; ++dx) {
+                for (int dy = -1; dy <= 1; ++dy) {
+                    for (int dz = -1; dz <= 1; ++dz) {
+                        if (dx != 0 || dy != 0 || dz != 0) {
+                            coords.add(original.add(dx, dy, dz));
+                        }
+                    }
+                }
+            }
+
+            return coords;
+        }
+
+        public static final class SwapCandidate implements Comparable<BlockSwapper.SwapCandidate> {
+            public final BlockPos coordinates;
+            public final int range;
+
+            public SwapCandidate(BlockPos coordinates, int range) {
+                this.coordinates = coordinates;
+                this.range = range;
+            }
+
+            public int compareTo(@Nonnull BlockSwapper.SwapCandidate other) {
+                return other.range - this.range;
+            }
+
+            public boolean equals(Object other) {
+                if (!(other instanceof BlockSwapper.SwapCandidate)) {
+                    return false;
+                } else {
+                    BlockSwapper.SwapCandidate cand = (BlockSwapper.SwapCandidate) other;
+                    return this.coordinates.equals(cand.coordinates) && this.range == cand.range;
+                }
+            }
+
+            public int hashCode() {
+                return Objects.hash(this.coordinates, this.range);
+            }
+        }
+    }
+
+    @Override
+    public void addInformation(ItemStack stack, World world, List<ITextComponent> list, ITooltipFlag flags) {
+        super.addInformation(stack, world, list, flags);
+        ITextComponent rank = new TranslationTextComponent("botania.rank" + getLevel(stack));
+        ITextComponent rankFormat = new TranslationTextComponent("botaniamisc.toolRank", rank);
+        list.add(rankFormat);
+        if (this.getMana(stack) == 2147483647) {
+            list.add((new TranslationTextComponent("botaniamisc.getALife")).func_240699_a_(TextFormatting.RED));
+        }
+    }
+
+    @Override
+    public float getManaFractionForDisplay(ItemStack stack) {
+        int level = getLevel(stack);
+        if (level <= 0) {
+            return getMana(stack) / (float) LEVELS[1];
+        } else if (level >= LEVELS.length - 1) {
+            return 1f;
+        } else {
+            return (getMana(stack) - LEVELS[level]) / (float) LEVELS[level + 1];
+        }
+    }
+
+    @Nonnull
+    @Override
+    public ActionResultType onItemUse(@Nonnull ItemUseContext ctx) {
+
+        if (ItemNBTHelper.getBoolean(ctx.getItem(), "hoemode", true)) {
+            // Terra AIOT has not hoe mode as there's no terra hoe and right click toggles AOE mining
```

Comment:
Hoe mode should be present.
For elementium and manasteel there is a hoe being added as well. Should be added for terrasteel, too. Maybe just tilt in 3x3 or 5x5 area. Shovel, too.